src/loginpage/RegistrationForm.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import styled from 'styled-components';
+import Input from '../components/Input';
+import Label from '../components/Label';
+import ColorInput from '../components/ColorInput';
+import SubmitButton from '../components/SubmitButton';
+
+const uid = require('uid');
+
+const StyledRegistrationForm = styled.form``;
+
+export default function RegistrationForm({
+  onNewUserGroup,
+  onLogin,
+  history,
+  userGroup
+}) {
+  function handleRegistration(event, onNewUserGroup, onLogin, history) {
+    event.preventDefault();
+    const userGroupId = uid();
+    const newAdminUser = {
+      id: uid(),
+      username: event.target.adminname.value,
+      userGroupname: event.target.userGroupName.value,
+      userGroupId,
+      email: event.target.email.value,
+      usercolor: event.target.usercolor.value,
+      role: 'admin'
+    };
+    const newUserGroup = {
+      name: event.target.userGroupName.value,
+      password: event.target.userGroupPassword.value,
+      id: userGroupId,
+      users: [newAdminUser]
+    };
+    onNewUserGroup(newUserGroup);
+    onLogin(newAdminUser, newUserGroup, history);
+  }
+
+  return (
+    <StyledRegistrationForm
+      onSubmit={event =>
+        handleRegistration(event, onNewUserGroup, onLogin, history)
+      }
+    >
+      <Label htmlFor="adminname" label="Gib deinen Namen an:" />
+      <Input name="adminname" required />
+      <Label htmlFor="email" label="Gib deine Email-Adresse an:" />
+      <Input name="email" required />
+      <Label htmlFor="usercolor" label="Wähle deine Farbe aus:" />
+      <ColorInput name="usercolor" userGroup={userGroup} />
+      <Label htmlFor="userGroupName" label="Gib deinen Familiennamen an:" />
+      <Input name="userGroupName" required />
+      <Label
+        htmlFor="userGroupPassword"
+        label="Gib dein Familienkennwort an:"
+      />
+      <Input name="userGroupPassword" required />
```

Comment:
same suggestion as above. you can think about creating an array with objects with `name` and `label` and map over it so you have only one or two lines here and outsource the array somewhere else